bash-example/deploy-file.sh

```diff
@@ -20,21 +20,49 @@ if [ ! -f "$localFilePath" ]; then
     exit 1
 fi
 
-which md5sum || exit $?
-which sha1sum || exit $?
+if command -v sha256sum > /dev/null && command -v sha1sum > /dev/null && command -v md5sum > /dev/null; then 
+# If the hashing tools provided by coreutils are installed then use them. They
+# give a convenient output that always starts with the hash. Thus, given that
+# we know how long each hash will be, we can just cut that bit off the start of
+# the string and we have our hash.
+    md5Value="$(md5sum "$localFilePath")"
+    md5Value="${md5Value:0:32}"
+    sha1Value="$(sha1sum "$localFilePath")"
+    sha1Value="${md5Value:0:40}"
+    sha256Value="$(sha256sum "$localFilePath")"
+    sha256Value="${sha256Value:0:65}"
+elif command -v openssl &> /dev/null; then
```

Comment:
OpenSSL is almost _always_ installed. SO it seems reasonable to add it as an alternative.